lib/SILGen/SILGenApply.cpp

```diff
@@ -2933,7 +2933,39 @@ static void emitDelayedArguments(SILGenFunction &SGF,
   }
 }
 
-static Expr *findStorageReferenceExprForBorrow(Expr *e) {
+namespace {
+/// Container to hold the result of a search for the storage reference
+/// when determining to emit a borrow.
+struct StorageRefResult {
+  // The direct storage reference.
+  Expr *storageRef;
+  // The root of the expression that accesses the storage in \c storageRef.
+  Expr *transitiveRoot;
```

Comment:
Can you add an example here of an AST and what would be the storage ref and what would be the transitiveRoot? Maybe add it to the top level comment on \p StorageRefResult?